test/unit_tests/event/preprocessor/ReadReceiptMiddlewareSpec.js

```diff
@@ -0,0 +1,75 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+// KARMA_SPECS=event/preprocessor/QuotedMessageMiddleware yarn test:app
+
+import ReadReceiptMiddleware from 'app/script/event/preprocessor/ReadReceiptMiddleware';
+
+describe('ReadReceiptMiddleware', () => {
+  const noop = () => {};
+  let readReceiptMiddleware;
+  const eventService = {loadEvent: noop, replaceEvent: noop};
+
+  beforeEach(() => {
+    readReceiptMiddleware = new ReadReceiptMiddleware(eventService);
+  });
+
+  describe('processEvent', () => {
+    it('Ignores read receipt for which original message is not found', () => {
+      spyOn(eventService, 'loadEvent').and.returnValue(Promise.resolve(undefined));
+      spyOn(eventService, 'replaceEvent');
+
+      const event = {
+        conversation: 'conversattionid',
```

Comment:
`conversattionid` -> `conversationid`

I would also put a real UUIDv4 there instead of `conversattionid` so that it looks more like an actual payload from the real world.